relayer/claim_request.go

```diff
@@ -37,30 +34,41 @@ func CreateRelayClaimRequest(
 	ctx context.Context,
 	claimerEthKey *ecdsa.PrivateKey,
 	ec *ethclient.Client,
-	swapCreatorAddr ethcommon.Address,
-	forwarderAddr ethcommon.Address,
-	swap *contracts.SwapCreatorSwap,
-	secret *[32]byte,
+	relaySwap *contracts.SwapCreatorRelaySwap,
+	secret [32]byte,
 ) (*message.RelayClaimRequest, error) {
-
-	signature, err := createForwarderSignature(
-		ctx,
+	signature, err := createRelayClaimSignature(
 		claimerEthKey,
-		ec,
-		swapCreatorAddr,
-		forwarderAddr,
-		swap,
-		secret,
+		relaySwap,
 	)
 	if err != nil {
 		return nil, err
 	}
 
 	return &message.RelayClaimRequest{
-		OfferID:         nil, // set elsewhere if sending to counterparty
-		SwapCreatorAddr: swapCreatorAddr,
-		Swap:            swap,
-		Secret:          secret[:],
-		Signature:       signature,
+		OfferID:   nil, // set elsewhere if sending to counterparty
+		RelaySwap: relaySwap,
+		Secret:    secret[:],
+		Signature: signature,
 	}, nil
 }
+
+func createRelayClaimSignature(
+	claimerEthKey *ecdsa.PrivateKey,
+	relaySwap *contracts.SwapCreatorRelaySwap,
+) ([]byte, error) {
+	signerAddress := ethcrypto.PubkeyToAddress(claimerEthKey.PublicKey)
+	if relaySwap.Swap.Claimer != signerAddress {
+		return nil, fmt.Errorf("signing key %s does not match claimer %s", signerAddress, relaySwap.Swap.Claimer)
+	}
+
+	msg := relaySwap.Hash()
+
+	// signature format is (r || s || v), v = 27/28
+	signature, err := rcommon.NewKeyFromPrivateKey(claimerEthKey).Sign(msg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign forward request digest: %w", err)
```

Comment:
update comment

also can probably remove the reliance on go-relayer, this is the only place it's used